test/fx/test_gradual_type.py

```diff
@@ -0,0 +1,44 @@
+import unittest
+import torch
+from torch.fx import symbolic_trace
+from torch.fx.tensor_type import Tensor_Type, Dyn
+from torch.fx.annotate import annotate
+
+
+class AnnotationsTest(unittest.TestCase):
+
+    def test_annotations(self):
+        """
+        Test type annotations in the forward function.
+        The annoation should appear in the n.graph
+        where n is the corresoinding node in the resulting graph.
+        """
+        class M(torch.nn.Module):
+            def forward(self, x: Tensor_Type((1, 2, 3, Dyn)), y: Dyn):
+                return torch.add(x, y)
+
+        module = M()
+        symbolic_traced : torch.fx.GraphModule = symbolic_trace(module)
+
+        for n in symbolic_traced.graph.nodes:
+            if n.name == 'x':
+                assert n.type == Tensor_Type((1, 2, 3, Dyn))
+            if n.name == 'y':
+                assert n.type == Dyn
```

Comment:
Node names aren't stable. I'd suggest checking this via the position of each placeholder node rather than by names, e.g.:

```
diff --git a/test/fx/test_gradual_type.py b/test/fx/test_gradual_type.py
index cddab0f004e..4a3276009ec 100644
--- a/test/fx/test_gradual_type.py
+++ b/test/fx/test_gradual_type.py
@@ -20,11 +20,15 @@ class AnnotationsTest(unittest.TestCase):
         module = M()
         symbolic_traced : torch.fx.GraphModule = symbolic_trace(module)
 
+        expected_ph_types = [
+            Tensor_Type((1, 2, 3, Dyn)),
+            Dyn,
+        ]
+        expected_iter = iter(expected_ph_types)
+
         for n in symbolic_traced.graph.nodes:
-            if n.name == 'x':
-                assert n.type == Tensor_Type((1, 2, 3, Dyn))
-            if n.name == 'y':
-                assert n.type == Dyn
+            if n.op == 'placeholder':
+                assert n.type == next(expected_iter)
 
     def test_annotate(self):
         class M(torch.nn.Module):
```